bolt/src/Passes/ReorderUncondJump.h

```diff
@@ -0,0 +1,57 @@
+//===--------- Passes/ReorderUncondJump.h ---------------------------------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+//
+//===----------------------------------------------------------------------===//
+
+#ifndef LLVM_TOOLS_LLVM_BOLT_PASSES_REORDERUNCONDJUMP_H
+#define LLVM_TOOLS_LLVM_BOLT_PASSES_REORDERUNCONDJUMP_H
+
+#include "BinaryPasses.h"
+
+// This pass founds cases when BBs have layout:
+// #BB1:
+// cmp
+// cond_jmp out
+// #BB2:
+// <code>
+// jmp #BB1
+// #BB3 <out>
+//
+// And swaps BB1 and BB2:
```

Comment:
Could you keep BB #'s the same as in the original (the way BOLT does it)? You'll have to introduce BB0.

Reply:
Sorry, you mean to change BB1 to BB0 & etc?

Reply:
Sorry, I mean treat BB#'s as BB names/labels, i.e. after the pass:

#BB0:
...
jmp #BB1
#BB2:
<loop body>
#BB1:
cmp ...
jncc #BB2
#BB3:
<loop exit>